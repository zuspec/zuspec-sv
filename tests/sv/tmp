sat
sat
(
  (comp_id #b11111111)
  (pcomp_id #b11111111)
  (r_inst_id #b00000000)
  (r_pool_id #b11111111)
)
sat
(
  (comp_id #b11111111)
  (pcomp_id #b11111111)
  (r_inst_id #b00000000)
  (r_pool_id #b11111111)
)
sat
(
  (comp_id #b11111111)
  (pcomp_id #b11111111)
  (r_inst_id #b00000000)
  (r_pool_id #b11111111)
)
sat
(
  (comp_id #b11111111)
  (pcomp_id #b11111111)
  (r_inst_id #b00000001)
  (r_pool_id #b11111111)
)
sat
(
  (comp_id #b11111111)
  (pcomp_id #b11111111)
  (r_inst_id #b00000001)
  (r_pool_id #b11111111)
)
sat
(
  (comp_id #b11111111)
  (pcomp_id #b11111111)
  (r_inst_id #b00000000)
  (r_pool_id #b11111111)
)
sat
(
  (comp_id #b11111111)
  (pcomp_id #b11111111)
  (r_inst_id #b00000000)
  (r_pool_id #b11111111)
)
sat
(
  (comp_id #b11111111)
  (pcomp_id #b11111111)
  (r_inst_id #b00000000)
  (r_pool_id #b11111110)
)
sat
(
  (comp_id #b11111111)
  (pcomp_id #b11111111)
  (r_inst_id #b00000001)
  (r_pool_id #b11111101)
)
sat
(
  (comp_id #b10001111)
  (pcomp_id #b00100000)
  (r_inst_id #b00000000)
  (r_pool_id #b10010000)
)
sat
(
  (comp_id #b11111111)
  (pcomp_id #b11111111)
  (r_inst_id #b00000000)
  (r_pool_id #b11111111)
)
sat
(
  (comp_id #b11111111)
  (pcomp_id #b11111111)
  (r_inst_id #b00000000)
  (r_pool_id #b11111101)
)
sat
(
  (comp_id #b11111111)
  (pcomp_id #b11111111)
  (r_inst_id #b00000000)
  (r_pool_id #b11111101)
)
sat
(
  (comp_id #b11111111)
  (pcomp_id #b11111111)
  (r_inst_id #b00000001)
  (r_pool_id #b11111101)
)
sat
(
  (comp_id #b11111111)
  (pcomp_id #b11111111)
  (r_inst_id #b00000001)
  (r_pool_id #b11111101)
)
