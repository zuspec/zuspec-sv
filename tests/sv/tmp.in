(set-logic QF_BV)
(check-sat)
(reset)
(set-option :produce-models true)
(set-logic QF_BV)
(define-fun __Vbv ((b Bool)) (_ BitVec 1) (ite b #b1 #b0))
(define-fun __Vbool ((v (_ BitVec 1))) Bool (= #b1 v))
(declare-fun A_comp_id () (_ BitVec 8))
(declare-fun A_pcomp_id () (_ BitVec 8))
(declare-fun A_r_inst_id () (_ BitVec 8))
(declare-fun A_r_pool_id () (_ BitVec 8))
(declare-fun B_comp_id () (_ BitVec 8))
(declare-fun B_pcomp_id () (_ BitVec 8))
(declare-fun B_r_inst_id () (_ BitVec 8))
(declare-fun B_r_pool_id () (_ BitVec 8))
(declare-fun C_comp_id () (_ BitVec 8))
(declare-fun C_pcomp_id () (_ BitVec 8))
(declare-fun C_r_inst_id () (_ BitVec 8))
(declare-fun C_r_pool_id () (_ BitVec 8))
(assert (= #b1 (__Vbv (=> (__Vbool (__Vbv (= ((_ zero_extend 24) A_pcomp_id) #x00000000))) (__Vbool (bvand (__Vbv (bvuge ((_ zero_extend 24) A_comp_id) #x00000001)) (__Vbv (bvule ((_ zero_extend 24) A_comp_id) #x00000002))))))))
(assert (= #b1 (__Vbv (=> (__Vbool (__Vbv (= ((_ zero_extend 24) A_comp_id) #x00000001))) (__Vbool (__Vbv (= ((_ zero_extend 24) A_r_pool_id) #x00000000)))))))
(assert (= #b1 (__Vbv (=> (__Vbool (__Vbv (= ((_ zero_extend 24) A_comp_id) #x00000002))) (__Vbool (__Vbv (= ((_ zero_extend 24) A_r_pool_id) #x00000001)))))))
(assert (= #b1 (bvand (__Vbv (bvuge ((_ zero_extend 24) A_r_inst_id) #x00000000)) (__Vbv (bvule ((_ zero_extend 24) A_r_inst_id) #x00000001)))))
(assert (= #b1 (__Vbv (=> (__Vbool (__Vbv (= ((_ zero_extend 24) B_pcomp_id) #x00000000))) (__Vbool (bvand (__Vbv (bvuge ((_ zero_extend 24) B_comp_id) #x00000001)) (__Vbv (bvule ((_ zero_extend 24) B_comp_id) #x00000002))))))))
(assert (= #b1 (__Vbv (=> (__Vbool (__Vbv (= ((_ zero_extend 24) B_comp_id) #x00000001))) (__Vbool (__Vbv (= ((_ zero_extend 24) B_r_pool_id) #x00000000)))))))
(assert (= #b1 (__Vbv (=> (__Vbool (__Vbv (= ((_ zero_extend 24) B_comp_id) #x00000002))) (__Vbool (__Vbv (= ((_ zero_extend 24) B_r_pool_id) #x00000001)))))))
(assert (= #b1 (bvand (__Vbv (bvuge ((_ zero_extend 24) B_r_inst_id) #x00000000)) (__Vbv (bvule ((_ zero_extend 24) B_r_inst_id) #x00000001)))))
(assert (= #b1 (__Vbv (=> (__Vbool (__Vbv (= ((_ zero_extend 24) C_pcomp_id) #x00000000))) (__Vbool (bvand (__Vbv (bvuge ((_ zero_extend 24) C_comp_id) #x00000001)) (__Vbv (bvule ((_ zero_extend 24) C_comp_id) #x00000002))))))))
(assert (= #b1 (__Vbv (=> (__Vbool (__Vbv (= ((_ zero_extend 24) C_comp_id) #x00000001))) (__Vbool (__Vbv (= ((_ zero_extend 24) C_r_pool_id) #x00000000)))))))
(assert (= #b1 (__Vbv (=> (__Vbool (__Vbv (= ((_ zero_extend 24) C_comp_id) #x00000002))) (__Vbool (__Vbv (= ((_ zero_extend 24) C_r_pool_id) #x00000001)))))))
(assert (= #b1 (bvand (__Vbv (bvuge ((_ zero_extend 24) C_r_inst_id) #x00000000)) (__Vbv (bvule ((_ zero_extend 24) C_r_inst_id) #x00000001)))))
(assert (= #b1 (__Vbv (= ((_ zero_extend 24) A_pcomp_id) #x00000000))))
(assert (= #b1 (__Vbv (= ((_ zero_extend 24) B_pcomp_id) #x00000000))))
(assert (= #b1 (__Vbv (= ((_ zero_extend 24) C_pcomp_id) #x00000000))))
(assert (= #b1 (__Vbv (not (__Vbool (bvand (__Vbv (= A_r_inst_id B_r_inst_id)) (__Vbv (= A_r_pool_id B_r_pool_id))))))))
(assert (= #b1 (__Vbv (not (__Vbool (bvand (__Vbv (= A_r_inst_id C_r_inst_id)) (__Vbv (= A_r_pool_id C_r_pool_id))))))))
(check-sat)
(get-value (A_comp_id A_pcomp_id A_r_inst_id A_r_pool_id B_comp_id B_pcomp_id B_r_inst_id B_r_pool_id C_comp_id C_pcomp_id C_r_inst_id C_r_pool_id ))
(assert (= #b0 (bvxor ((_ extract 2 2) A_comp_id) ((_ extract 4 4) A_comp_id) ((_ extract 0 0) A_pcomp_id) ((_ extract 2 2) A_pcomp_id) ((_ extract 5 5) A_pcomp_id) ((_ extract 6 6) A_pcomp_id) ((_ extract 2 2) A_r_inst_id) ((_ extract 3 3) A_r_inst_id) ((_ extract 5 5) A_r_inst_id) ((_ extract 6 6) A_r_inst_id) ((_ extract 7 7) A_r_inst_id) ((_ extract 0 0) A_r_pool_id) ((_ extract 1 1) A_r_pool_id) ((_ extract 3 3) A_r_pool_id) ((_ extract 6 6) A_r_pool_id) ((_ extract 7 7) A_r_pool_id) ((_ extract 1 1) B_comp_id) ((_ extract 3 3) B_comp_id) ((_ extract 4 4) B_comp_id) ((_ extract 6 6) B_comp_id) ((_ extract 0 0) B_pcomp_id) ((_ extract 2 2) B_pcomp_id) ((_ extract 4 4) B_pcomp_id) ((_ extract 6 6) B_pcomp_id) ((_ extract 7 7) B_pcomp_id) ((_ extract 1 1) B_r_inst_id) ((_ extract 2 2) B_r_inst_id) ((_ extract 5 5) B_r_inst_id) ((_ extract 6 6) B_r_inst_id) ((_ extract 7 7) B_r_inst_id) ((_ extract 0 0) B_r_pool_id) ((_ extract 1 1) B_r_pool_id) ((_ extract 4 4) B_r_pool_id) ((_ extract 0 0) C_comp_id) ((_ extract 3 3) C_comp_id) ((_ extract 4 4) C_comp_id) ((_ extract 3 3) C_pcomp_id) ((_ extract 7 7) C_pcomp_id) ((_ extract 2 2) C_r_inst_id) ((_ extract 3 3) C_r_inst_id) ((_ extract 4 4) C_r_inst_id) ((_ extract 5 5) C_r_inst_id) ((_ extract 6 6) C_r_inst_id) ((_ extract 7 7) C_r_inst_id) ((_ extract 1 1) C_r_pool_id) ((_ extract 3 3) C_r_pool_id) ((_ extract 5 5) C_r_pool_id) ((_ extract 7 7) C_r_pool_id))))

(check-sat)
(get-value (A_comp_id A_pcomp_id A_r_inst_id A_r_pool_id B_comp_id B_pcomp_id B_r_inst_id B_r_pool_id C_comp_id C_pcomp_id C_r_inst_id C_r_pool_id ))
(assert (= #b1 (bvxor ((_ extract 0 0) A_comp_id) ((_ extract 4 4) A_comp_id) ((_ extract 7 7) A_comp_id) ((_ extract 0 0) A_pcomp_id) ((_ extract 3 3) A_pcomp_id) ((_ extract 5 5) A_pcomp_id) ((_ extract 7 7) A_pcomp_id) ((_ extract 0 0) A_r_inst_id) ((_ extract 2 2) A_r_inst_id) ((_ extract 3 3) A_r_inst_id) ((_ extract 6 6) A_r_inst_id) ((_ extract 7 7) A_r_inst_id) ((_ extract 2 2) A_r_pool_id) ((_ extract 3 3) A_r_pool_id) ((_ extract 4 4) A_r_pool_id) ((_ extract 5 5) A_r_pool_id) ((_ extract 3 3) B_comp_id) ((_ extract 4 4) B_comp_id) ((_ extract 5 5) B_comp_id) ((_ extract 6 6) B_comp_id) ((_ extract 0 0) B_pcomp_id) ((_ extract 1 1) B_pcomp_id) ((_ extract 4 4) B_pcomp_id) ((_ extract 6 6) B_pcomp_id) ((_ extract 7 7) B_pcomp_id) ((_ extract 1 1) B_r_inst_id) ((_ extract 2 2) B_r_inst_id) ((_ extract 4 4) B_r_inst_id) ((_ extract 1 1) B_r_pool_id) ((_ extract 3 3) B_r_pool_id) ((_ extract 4 4) B_r_pool_id) ((_ extract 7 7) B_r_pool_id) ((_ extract 2 2) C_comp_id) ((_ extract 6 6) C_comp_id) ((_ extract 7 7) C_comp_id) ((_ extract 0 0) C_pcomp_id) ((_ extract 1 1) C_pcomp_id) ((_ extract 7 7) C_pcomp_id) ((_ extract 0 0) C_r_inst_id) ((_ extract 4 4) C_r_inst_id) ((_ extract 5 5) C_r_inst_id) ((_ extract 6 6) C_r_inst_id) ((_ extract 0 0) C_r_pool_id) ((_ extract 1 1) C_r_pool_id) ((_ extract 3 3) C_r_pool_id) ((_ extract 4 4) C_r_pool_id) ((_ extract 6 6) C_r_pool_id) ((_ extract 7 7) C_r_pool_id))))

(check-sat)
(get-value (A_comp_id A_pcomp_id A_r_inst_id A_r_pool_id B_comp_id B_pcomp_id B_r_inst_id B_r_pool_id C_comp_id C_pcomp_id C_r_inst_id C_r_pool_id ))
(assert (= #b0 (bvxor ((_ extract 1 1) A_comp_id) ((_ extract 7 7) A_comp_id) ((_ extract 1 1) A_pcomp_id) ((_ extract 2 2) A_pcomp_id) ((_ extract 3 3) A_pcomp_id) ((_ extract 7 7) A_pcomp_id) ((_ extract 3 3) A_r_inst_id) ((_ extract 4 4) A_r_inst_id) ((_ extract 5 5) A_r_inst_id) ((_ extract 7 7) A_r_inst_id) ((_ extract 1 1) A_r_pool_id) ((_ extract 3 3) A_r_pool_id) ((_ extract 4 4) A_r_pool_id) ((_ extract 5 5) A_r_pool_id) ((_ extract 6 6) A_r_pool_id) ((_ extract 7 7) A_r_pool_id) ((_ extract 1 1) B_comp_id) ((_ extract 2 2) B_comp_id) ((_ extract 3 3) B_comp_id) ((_ extract 4 4) B_comp_id) ((_ extract 6 6) B_comp_id) ((_ extract 7 7) B_comp_id) ((_ extract 0 0) B_pcomp_id) ((_ extract 2 2) B_pcomp_id) ((_ extract 1 1) B_r_inst_id) ((_ extract 3 3) B_r_inst_id) ((_ extract 3 3) B_r_pool_id) ((_ extract 4 4) B_r_pool_id) ((_ extract 6 6) B_r_pool_id) ((_ extract 1 1) C_comp_id) ((_ extract 2 2) C_comp_id) ((_ extract 3 3) C_comp_id) ((_ extract 1 1) C_pcomp_id) ((_ extract 4 4) C_pcomp_id) ((_ extract 5 5) C_pcomp_id) ((_ extract 6 6) C_pcomp_id) ((_ extract 0 0) C_r_inst_id) ((_ extract 1 1) C_r_inst_id) ((_ extract 3 3) C_r_inst_id) ((_ extract 5 5) C_r_inst_id) ((_ extract 6 6) C_r_inst_id) ((_ extract 7 7) C_r_inst_id) ((_ extract 1 1) C_r_pool_id) ((_ extract 2 2) C_r_pool_id) ((_ extract 4 4) C_r_pool_id) ((_ extract 5 5) C_r_pool_id) ((_ extract 6 6) C_r_pool_id) ((_ extract 7 7) C_r_pool_id))))

(check-sat)
(get-value (A_comp_id A_pcomp_id A_r_inst_id A_r_pool_id B_comp_id B_pcomp_id B_r_inst_id B_r_pool_id C_comp_id C_pcomp_id C_r_inst_id C_r_pool_id ))
(assert (= #b0 (bvxor ((_ extract 0 0) A_comp_id) ((_ extract 2 2) A_comp_id) ((_ extract 3 3) A_comp_id) ((_ extract 4 4) A_comp_id) ((_ extract 5 5) A_comp_id) ((_ extract 6 6) A_comp_id) ((_ extract 7 7) A_comp_id) ((_ extract 3 3) A_pcomp_id) ((_ extract 5 5) A_pcomp_id) ((_ extract 6 6) A_pcomp_id) ((_ extract 1 1) A_r_inst_id) ((_ extract 3 3) A_r_inst_id) ((_ extract 5 5) A_r_inst_id) ((_ extract 2 2) A_r_pool_id) ((_ extract 3 3) A_r_pool_id) ((_ extract 4 4) A_r_pool_id) ((_ extract 5 5) A_r_pool_id) ((_ extract 7 7) A_r_pool_id) ((_ extract 0 0) B_comp_id) ((_ extract 2 2) B_comp_id) ((_ extract 6 6) B_comp_id) ((_ extract 1 1) B_pcomp_id) ((_ extract 4 4) B_pcomp_id) ((_ extract 5 5) B_pcomp_id) ((_ extract 6 6) B_pcomp_id) ((_ extract 1 1) B_r_inst_id) ((_ extract 2 2) B_r_inst_id) ((_ extract 3 3) B_r_inst_id) ((_ extract 0 0) B_r_pool_id) ((_ extract 1 1) B_r_pool_id) ((_ extract 4 4) B_r_pool_id) ((_ extract 5 5) B_r_pool_id) ((_ extract 7 7) B_r_pool_id) ((_ extract 0 0) C_comp_id) ((_ extract 1 1) C_comp_id) ((_ extract 6 6) C_comp_id) ((_ extract 7 7) C_comp_id) ((_ extract 0 0) C_pcomp_id) ((_ extract 4 4) C_pcomp_id) ((_ extract 1 1) C_r_inst_id) ((_ extract 3 3) C_r_inst_id) ((_ extract 4 4) C_r_inst_id) ((_ extract 5 5) C_r_inst_id) ((_ extract 6 6) C_r_inst_id) ((_ extract 0 0) C_r_pool_id) ((_ extract 2 2) C_r_pool_id) ((_ extract 3 3) C_r_pool_id) ((_ extract 5 5) C_r_pool_id))))

(check-sat)
(get-value (A_comp_id A_pcomp_id A_r_inst_id A_r_pool_id B_comp_id B_pcomp_id B_r_inst_id B_r_pool_id C_comp_id C_pcomp_id C_r_inst_id C_r_pool_id ))
