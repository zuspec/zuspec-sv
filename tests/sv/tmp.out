sat
sat
(
  (A_comp_id #b00000010)
  (A_pcomp_id #b00000000)
  (A_r_inst_id #b00000000)
  (A_r_pool_id #b00000001)
  (B_comp_id #b00000010)
  (B_pcomp_id #b00000000)
  (B_r_inst_id #b00000001)
  (B_r_pool_id #b00000001)
  (C_comp_id #b00000010)
  (C_pcomp_id #b00000000)
  (C_r_inst_id #b00000001)
  (C_r_pool_id #b00000001)
)
sat
(
  (A_comp_id #b00000001)
  (A_pcomp_id #b00000000)
  (A_r_inst_id #b00000001)
  (A_r_pool_id #b00000000)
  (B_comp_id #b00000010)
  (B_pcomp_id #b00000000)
  (B_r_inst_id #b00000001)
  (B_r_pool_id #b00000001)
  (C_comp_id #b00000010)
  (C_pcomp_id #b00000000)
  (C_r_inst_id #b00000001)
  (C_r_pool_id #b00000001)
)
sat
(
  (A_comp_id #b00000010)
  (A_pcomp_id #b00000000)
  (A_r_inst_id #b00000000)
  (A_r_pool_id #b00000001)
  (B_comp_id #b00000010)
  (B_pcomp_id #b00000000)
  (B_r_inst_id #b00000001)
  (B_r_pool_id #b00000001)
  (C_comp_id #b00000001)
  (C_pcomp_id #b00000000)
  (C_r_inst_id #b00000001)
  (C_r_pool_id #b00000000)
)
sat
(
  (A_comp_id #b00000010)
  (A_pcomp_id #b00000000)
  (A_r_inst_id #b00000001)
  (A_r_pool_id #b00000001)
  (B_comp_id #b00000010)
  (B_pcomp_id #b00000000)
  (B_r_inst_id #b00000000)
  (B_r_pool_id #b00000001)
  (C_comp_id #b00000001)
  (C_pcomp_id #b00000000)
  (C_r_inst_id #b00000000)
  (C_r_pool_id #b00000000)
)
sat
(
  (A_comp_id #b00000010)
  (A_pcomp_id #b00000000)
  (A_r_inst_id #b00000001)
  (A_r_pool_id #b00000001)
  (B_comp_id #b00000010)
  (B_pcomp_id #b00000000)
  (B_r_inst_id #b00000000)
  (B_r_pool_id #b00000001)
  (C_comp_id #b00000001)
  (C_pcomp_id #b00000000)
  (C_r_inst_id #b00000000)
  (C_r_pool_id #b00000000)
)
